edgraph/server.go

```diff
@@ -1907,3 +2087,51 @@ func isDropAll(op *api.Operation) bool {
 	}
 	return false
 }
+
+func verifyUniqueWithinMutation(qc *queryContext) error {
+	if len(qc.uniqueVars) == 0 {
+		return nil
+	}
+
+	for i := range qc.uniqueVars {
+		gmuIndex, rdfIndex := decodeIndex(i)
+		if len(qc.gmuList[gmuIndex].Set) == 0 {
+			return nil
+		}
+		pred1 := qc.gmuList[gmuIndex].Set[rdfIndex]
+		pred1Value := dql.TypeValFrom(pred1.ObjectValue).Value
+		for j := range qc.uniqueVars {
+			gmuIndex2, rdfIndex2 := decodeIndex(j)
+			pred2 := qc.gmuList[gmuIndex2].Set[rdfIndex2]
+			if pred2.Predicate == pred1.Predicate && dql.TypeValFrom(pred2.ObjectValue).Value == pred1Value &&
+				pred2.Subject != pred1.Subject {
+				return errors.Errorf("could not insert duplicate value [%v] for predicate [%v]",
+					dql.TypeValFrom(pred1.ObjectValue).Value, pred1.Predicate)
+			}
+		}
+	}
+	return nil
+}
+
+// encodeBit two uint32 numbers by bit.
+// First 32 bits store k1 and last 32 bits store k2.
+func encodeIndex(k1, k2 uint32) uint64 {
```

Comment:
put encodeIndex and decodeIndex close to each other and fix the comments. The comment still says encodeBit.

Reply:
I am also thinking that we should take int as an argument and check whether the conversion to uint32 is possible or not.